src/public/modules/forms/helpers/CsvMergedHeadersGenerator.js

```diff
@@ -107,21 +107,25 @@ class CsvMergedHeadersGenerator extends CsvGenerator {
 
     // Craft a new csv row for each unprocessed record
     // O(qn), where q = number of unique questions, n = number of submissions.
-    this.unprocessed.forEach((up) => {
-      let createdAt = moment(up.created).tz('Asia/Singapore')
-      createdAt = createdAt.isValid()
-        ? createdAt.format('DD MMM YYYY hh:mm:ss A')
-        : createdAt
-      let row = [up.submissionId, createdAt]
-      for (let [fieldId] of this.fieldIdToQuestion) {
-        const numCols = this.fieldIdToNumCols[fieldId]
-        for (let colIndex = 0; colIndex < numCols; colIndex++) {
-          row.push(this._extractAnswer(up.record, fieldId, colIndex))
+    this.unprocessed
+      .map((up) => {
+        const createdAt = moment(up.created).tz('Asia/Singapore')
+        return { up, createdAt }
```

Comment:
suggest moving this parsing into the `_dateComparator` instead, so that the __dateComparator_ can compare instances of `a` and `b` directly.

also - what is the type of `up.created`? is it possible that the cast to `Asia/Singapore` is unnecessary?

Reply:
I've made the changes! However, I only passed in `up.created` into `_dateComparator` instead of `up`, as I believe `_dateComparator` should only be handling dates. 

`up.created` is a `string` so I think the cast is still necessary.